### PR TITLE
Enable more rubocop rules

### PR DESCRIPTION
**Pull request by duckinator**
# Description:

I enabled more RuboCop rules, and ran `rubocop --auto-correct` with each change as @deivid-rodriguez suggested in #2407.

This PR replaces #2407.

# Tasks:

- [x] Describe the problem / feature
- [x] Write code to solve the problem
- [ ] Get code review from coworkers / friends

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -5,9 +5,42 @@ AllCops:
...
+Layout/EndOfLine:
+  Enabled: true
+  EnforcedStyle: lf
+
+Layout/IndentHash:
+  Enabled: true
+  EnforcedStyle: consistent
+
+# NOTE: When used with --auto-correct, this only seems to auto-correct the
+#       first array item for some reason.
```
Comment (deivid-rodriguez) [inline .rubocop.yml]: Yeah, I noticed this when I reviewed the other PR. See https://github.com/rubygems/rubygems/pull/2407#discussion_r221091861. Given this limitation, I don't think it makes sense to use `Layout/IndentArray` but not `Layout/AlignArray`, so could we enable `Layout/AlignArray` as well?

File: .rubocop.yml [hunk 2]
```diff
@@ -5,9 +5,42 @@ AllCops:
...
+Layout/IndentArray:
+  Enabled: true
+  EnforcedStyle: consistent
+  Exclude:
+    # This is a generated file.
+    - lib/rubygems/util/licenses.rb
```
Comment (deivid-rodriguez) [inline .rubocop.yml]: I think we can fix this at the source by removing spaces from [here](https://github.com/rubygems/rubygems/blob/cc881a806afc90731d5a912415aa945726978b54/util/generate_spdx_license_list.rb#L28) and [here](https://github.com/rubygems/rubygems/blob/cc881a806afc90731d5a912415aa945726978b54/util/generate_spdx_license_list.rb#L33).

Comment (duckinator) [reply .rubocop.yml]: Given the size of this PR, I'd like to make a second follow-up PR to address this one after this gets merged, if that's okay with other folks.

File: .rubocop.yml (hunk 1 above)
Comment (duckinator) [reply .rubocop.yml]: I thought there was something like that, but couldn't find it. Thanks for the info!

File: .rubocop.yml (hunk 2 above)
Comment (deivid-rodriguez) [reply .rubocop.yml]: Sounds good to me! :+1:

File: .rubocop.yml (hunk 1 above)
Comment (duckinator) [reply .rubocop.yml]: Attempting to enable AlignArray revealed a potential bug, or at the very least some questionable code lol https://github.com/rubygems/rubygems/issues/2437

**Comment by colby-swandale** [comment]
Great! @bundlerbot r+

**Comment by colby-swandale** [comment]
@bundlerbot r+

**Pull request merged**